site/source/docs/tools_reference/emcc.rst

```diff
@@ -107,69 +111,42 @@ Options that are modified or new in *emcc* are listed below:
 ``-g``
   Preserve debug information.
 
-  - When compiling to bitcode, this is the same as in *Clang* and *gcc* (it adds debug information to the object files).
-  - When compiling from source to JavaScript or bitcode to JavaScript, it is equivalent to :ref:`-g3 <emcc-g3>` (discards LLVM debug info including C/C++ line numbers, but otherwise keeps as much debug information as possible). Use :ref:`-g4 <emcc-g4>` to get line number debugging information in JavaScript.
+  - When compiling to object files, this is the same as in *Clang* and *gcc* (it adds debug information to the object files).
+  - This is equivalent to :ref:`-g3 <emcc-g3>`.
 
 .. _emcc-gN:
 
 ``-g<level>``
-  Controls how much debug information is kept when compiling from bitcode to JavaScript. Each level builds on the previous level:
+  Controls the level of debuggability. Each level builds on the previous one:
```

Comment:
These descriptions only apply for link time, right?

Reply:
Mostly link, but `-g3` and `-g4` are the same as `-g` for link. I clarified in each one exactly when it affects things.